### PR TITLE
SNOW-910685: Support creating session from default configuration file 

### PR DESCRIPTION
**Pull request by sfc-gh-jdu**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-910685

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Bump the connector version of connector so we have the latest change of default config file, and support reading connection parameters from default config file for creating session, and add a test

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@
 
 - Added support for specifying clustering keys when saving dataframes using `DataFrame.save_as_table`.
 - Accept `Iterable` objects input for `schema` when creating dataframes using `Session.create_dataframe`.
+- Added support for creating a Snowpark session from a configuration file or environment variables.
```
Comment (sfc-gh-jdu) [inline CHANGELOG.md:9]: I think we need more documentation for this feature, currently I can only find it in connector's changelog. So I created https://snowflakecomputing.atlassian.net/browse/SNOW-910845, and we should do it before next snowpark release.

File: tests/integ/test_session.py [hunk 2]
```diff
@@ -550,3 +550,26 @@ def test_sql_simplifier_disabled_on_session(db_parameters):
...
+        with monkeypatch.context() as m:
+            m.setenv("SNOWFLAKE_CONNECTIONS", tomlkit.dumps(doc))
+            m.setenv("SNOWFLAKE_DEFAULT_CONNECTION_NAME", "default")
+            with Session.builder.create() as new_session:
+                _ = new_session.sql("select 1").collect()[0][0]
+    except Exception:
+        # This is my way of guaranteeing that we'll not expose the
+        # sensitive information that this test needs to handle.
+        # db_parameter contains passwords.
+        pytest.fail("something failed", pytrace=False)
```
Comment (sfc-gh-yixie) [inline tests/integ/test_session.py:575]: Pls create a JIRA to update here to have the full stack trace.

**Review by sfc-gh-yixie — APPROVED** [review]

File: tests/integ/test_session.py (hunk 2 above)
Comment (sfc-gh-jdu) [reply tests/integ/test_session.py:575]: created https://snowflakecomputing.atlassian.net/browse/SNOW-910886

File: src/snowflake/snowpark/session.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
 import pkg_resources
 
 from snowflake.connector import ProgrammingError, SnowflakeConnection
+from snowflake.connector.config_manager import _get_default_connection_params
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/session.py:24]: Will this import work in stored proc?

Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/session.py:24]: Sorry I didn't catch your message here 🤦 yea it won't work actually.... I'll put out another PR to fix it

**Pull request merged**